### PR TITLE
Thread safety improvements

### PR DESCRIPTION
**Pull request by phlptp**
This branch is going to be a longer running branch.  This pull request is the first in what will likely be a series of progressive steps that build on eachother to put some effort into cleaning up the core code base, make it a little easier to maintain and improve the thread safety.  But given the changes will build on eachother and there will be breaks between them I thought it wise to do more frequent pull requests to for the incremental improvements.  

comments and suggestions are welcome.

### CODE REVIEW

**Comment by nightlark** [comment]
This looks like it can be merged in its current state. There's a few warnings about C++ style // comments in some of the C interface examples -- looks like code that was just commented out and left in that could maybe be removed.

**Comment by phlptp** [comment]
I wasn't entirely sure why those warnings showed up now since I didn't change that code.  In any case the next pull request coming after this one which is an update of some of Json configuration stuff, fixes those warnings.    I think they showed up when we merged the C++98 stuff which had some extra C flags, but we would have to go back and look at all the logs to determine exactly.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+<img src="docs/img/HELICS_Logo.png" width="400">
 </p>
 
-
 Repository for the source code for the Hierarchical Engine for Large-scale Infrastructure Co-Simulation (HELICS) tool.
 
 # Install Instructions
 
-[Windows](./installing-on-windows.md)
+[Windows](docs/installing-on-windows.md)
```
Comment (kdheepak) [inline README.md]: I made a similar change on master for cleaning up the root directory. Feel free to merge this change in, I'll fix the conflict when I am merging to master eventually.

Comment (phlptp) [reply README.md]: Lets merge this for now and then I will merge master into it before doing the PR into develop.  Then it should be clean when we eventually merge with Master.  Maybe next week for an alpha2 release.

**Comment by phlptp** [comment]
Unless there are some objections and change requests I will merge this later this afternoon.

**Pull request merged**